crates/nh-core/src/ui.rs

```
@@ -0,0 +1,64 @@
+  Paint::new(ICON_DIVIDER.repeat(width)).fg(GREY).to_string()
+}
+
+/// Available (free) disk space in bytes for the filesystem containing `path`.
+/// Returns None if statvfs is unsupported or fails.
+#[cfg(unix)]
+#[must_use]
+pub fn available_space_bytes(path: &str) -> Option<u64> {
+  use std::path::Path;
+  nix::sys::statvfs::statvfs(Path::new(path))
+    .ok()
+    .map(|s| u64::from(s.blocks_available()) * s.fragment_size())
```

Clippy (1.97.1)

[warning] useless conversion to the same type

(warning)